tools/bumper/component_commands.go

```diff
@@ -216,14 +215,21 @@ func (componentOps *gitComponent) getLatestTaggedFromBranch(repo, owner, branch,
 			return "", "", errors.Wrap(err, "Failed to get commit-sha from tag name")
 		}
 
-		for _, commit := range branchCommits {
-			if commit.GetSHA() == commitShaOfTag {
-				return tagName, commit.GetSHA(), nil
+		for pageIdx := 1; pageIdx <= maxPagePaginate; pageIdx++ {
+			branchCommits, _, err := componentOps.githubInterface.listCommits(owner, repo, &github.CommitsListOptions{ListOptions: github.ListOptions{PerPage: pageSize, Page: pageIdx}, SHA: branch})
```

Comment:
if there are only 3 pages, and you try to fetch page 4 it will fail
please make sure the page exists, there is a next_page field for that

Reply:
I can't find this field. Can you direct me to it?

Reply:
I think via the link header
"The URL for the last page is followed by rel="last"."
https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28

you can also check if err is page not exists / overflow, but it is less nice of course

Reply:
```
		if resp.NextPage == 0 {
			break
		}
```

resp is the 2nd return value of listCommits

Reply:
I checked, it works